compiler/rustc_expand/src/config.rs

```diff
@@ -310,15 +310,13 @@ impl<'a> StripUnconfigured<'a> {
                     Some((AttrAnnotatedTokenTree::Delimited(sp, delim, inner), *spacing))
                         .into_iter()
                 }
+                AttrAnnotatedTokenTree::Token(
+                    ref token @ Token { kind: TokenKind::Interpolated(ref nt), .. },
```

Comment:
Nit: it is exactly to avoid having to write out a pattern such as this is why people tend to write code in the way as it was before.

I wonder though if

```rust
AttrAnnotatedTokenTree::Token(token) if let TokenKind::Interpolated(nt) = token.kind => { ... }
```

would work?